test/object-store/sync/flx_migration.cpp

```diff
@@ -537,7 +537,9 @@ TEST_CASE("An interrupted migration or rollback can recover on the next session"
         return [&config, &state, error](std::weak_ptr<SyncSession> weak_session,
                                         const SyncClientHookData& data) mutable {
             auto session = weak_session.lock();
-            REQUIRE(session);
+            if (!session) {
```

Comment:
why do we need this change?

Reply:
To address an annoying TSAN error with the Catch2 library - what I was talking about during our sync meeting this morning.

Reply:
got it